### PR TITLE
[MSPAINT] Refactor loading/saving image file

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

JIRA issue: [CORE-16982](https://jira.reactos.org/browse/CORE-16982), [CORE-16979](https://jira.reactos.org/browse/CORE-16979)

- Add `DoLoadImageFile`, `CreateWhiteDIB`, and `CMainWindow::ConfirmSave` functions.
- Replace `LoadDIBFromFile` function with `DoLoadImageFile`.
- Fix the empty file loading.
- Delete `UpdateApplicationProperties` function.
- Rewrite `SaveDIBToFile` function.

### CODE REVIEW

**Comment by KyleKatarn31** [comment]
Thanks for looking at my ticket.

I'll edit the ticket because i've found another weakness of Paint load file function.

Procedure :
- Take iexplore.exe and rename it iexplore.bmp
- in Paint File>Open ==> iexplore.bmp
- Nothing happens (no error presented to the user)

**Comment by KyleKatarn31** [comment]
> 
> ## Purpose
> 
> JIRA issue: [CORE-16979](https://jira.reactos.org/browse/CORE-16979)
> 
>     * Add `DoLoadImageFile` function.
> 
>     * Replace `LoadDIBFromFile` function with `DoLoadImageFile`.
> 
>     * Fix the empty file loading.
> 
>     * Delete `UpdateApplicationProperties` function.

Please replace **MSPaint** by **Paint** in the PR title.

**Comment by katahiromz** [comment]
fixed

File: base/applications/mspaint/dib.cpp [hunk 1]
```diff
@@ -82,41 +82,111 @@ void ShowFileLoadError(LPCTSTR name)
...
+        }
+    }
+
+    CImage img;
+    img.Load(name);
+    hBitmap = img.Detach();
+
+    if (hBitmap == NULL)
     {
+        // cannot open and not empty
```
Comment (KyleKatarn31) [inline base/applications/mspaint/dib.cpp:138]: is comment still relevant for "empty" now that there is a dwFileSize == 0 check above ?

**Comment by KyleKatarn31** [comment]
> 
> 
> fixed

Good ! Please add https://jira.reactos.org/browse/CORE-16982 reference to the PR title as the scope also covers this one on top of https://jira.reactos.org/browse/CORE-16979

**Comment by binarymaster** [comment]
> Please replace MSPaint by Paint in the PR title.

WTF @KyleKatarn31 ? Module name is MSPAINT, not Paint.

**Comment by KyleKatarn31** [comment]
> 
> 
> > Please replace MSPaint by Paint in the PR title.
> 
> WTF @KyleKatarn31 ? Module name is MSPAINT, not Paint.

My mistake, sorry.

File: base/applications/mspaint/dib.cpp (hunk 1 above)
Comment (katahiromz) [reply base/applications/mspaint/dib.cpp]: OK.

**Comment by katahiromz** [comment]
Ready to go.

File: base/applications/mspaint/winproc.cpp [hunk 2]
```diff
@@ -174,17 +159,15 @@ void CMainWindow::InsertSelectionFromHBITMAP(HBITMAP bitmap, HWND window)
...
-    if (bmNew != NULL)
-    {
-        UpdateApplicationProperties(bmNew, droppedfile);
-    }
+
+    HDROP hDrop = (HDROP)wParam;
+    DragQueryFile(hDrop, 0, droppedfile, SIZEOF(droppedfile));
+    DragFinish(hDrop);
+
+    if (ConfirmSave())
```
Comment (roybaer) [inline base/applications/mspaint/winproc.cpp:168]: Nitpick: You're using && later on. Why not here?

Comment (katahiromz) [reply base/applications/mspaint/winproc.cpp]: Used `&&`.

File: base/applications/mspaint/dib.cpp [hunk 3]
```diff
@@ -43,34 +66,29 @@ GetDIBHeight(HBITMAP hBitmap)
...
     img.Attach(hBitmap);
     img.Save(FileName);  // TODO: error handling
     img.Detach();
 
-    // update time and size
-
-    HANDLE hFile =
-        CreateFile(FileName, GENERIC_READ, FILE_SHARE_READ, NULL, OPEN_EXISTING, FILE_FLAG_SEQUENTIAL_SCAN, NULL);
-    if (hFile == INVALID_HANDLE_VALUE)
+    WIN32_FIND_DATAW find;
```
Comment (katahiromz) [inline base/applications/mspaint/dib.cpp:76]: WIN32_FIND_DATA

File: base/applications/mspaint/dib.cpp [hunk 4]
```diff
@@ -43,34 +66,29 @@ GetDIBHeight(HBITMAP hBitmap)
...
-    if (time)
-    {
-        FILETIME ft;
-        GetFileTime(hFile, NULL, NULL, &ft);
-        FileTimeToSystemTime(&ft, time);
-    }
-    if (size)
-        *size = GetFileSize(hFile, NULL);
+    // update time and size
+    FileTimeToSystemTime(&find.ftLastWriteTime, &fileTime);
```
Comment (katahiromz) [inline base/applications/mspaint/dib.cpp:83]: Convert to local.

File: base/applications/mspaint/dib.cpp [hunk 5]
```diff
@@ -82,41 +100,120 @@ void ShowFileLoadError(LPCTSTR name)
     mainWindow.MessageBox(strText, strProgramName, MB_OK | MB_ICONEXCLAMATION);
 }
 
-void
-LoadDIBFromFile(HBITMAP * hBitmap, LPCTSTR name, LPSYSTEMTIME time, int *size, int *hRes, int *vRes)
+BOOL SetBitmapAndInfo(HBITMAP *phBitmap, LPCTSTR name, DWORD dwFileSize, BOOL isFile)
```
Comment (katahiromz) [inline base/applications/mspaint/dib.cpp:103]: Make *phBitmap the return value

File: base/applications/mspaint/dib.cpp [hunk 6]
```diff
@@ -82,41 +100,120 @@ void ShowFileLoadError(LPCTSTR name)
...
+    isAFile = isFile;
+    if (isAFile)
+        registrySettings.SetMostRecentFile(filepathname);
+
+    imageSaved = TRUE;
+
+    return TRUE;
+}
+
+BOOL DoLoadImageFile(HWND hwnd, HBITMAP *phBitmap, LPCTSTR name, BOOL fIsMainFile)
```
Comment (katahiromz) [inline base/applications/mspaint/dib.cpp:158]: Make  *phBitmap the return value

**Pull request merged**